private/numerics/functions/bessel.rkt

```diff
@@ -0,0 +1,367 @@
+                       z²
+                       +.1290918471896187879332737e+5
+                       +.1306678308784402036110575e+5
+                       +.310928141677002883350924e+4
+                       +.16904721775008609992033e+3
+                       +.1e+1))]
+               [q₁ (/ (poly-by-coeffs->value
+                       z²
+                       ;This line or the next is in error
+                       ;+.14465282874995208675225e+3   
+                       +.14465282874995208765225e+3
+                       +.1744291689092425885102e+3
+                       +.5173653281836591636536e+2
+                       +.379944537969806734901e+1
+                       +.36363466476034710809e-1)
+                      (poly-by-coeffs->value
+                       z²
+                       +.308592701333231723110639e+4
+                       +.373434010601630179517765e+4
+                       +.11191098527047487025919e+4
+                       +.8522392064341340397334e+2
+                       +.1e+1))]
+               [ans (* (sqrt (/ 2/π ax))
+                       (- (* (cos ax-3π/4) p₁)
+                          (* z (sin ax-3π/4) q₁)))])
+          (if (< x 0.0) (- ans) ans)))))
+
+(define (bessj n x)
+  (define acc 40)
+  (define bigno 1.0e10)
+  (define bigni 1.0e-10)
+  (cond
+   [(fx= n 0) (bessj₀ x)]
+   [(fx= n 1) (bessj₁ x)]
+   [(= x 0.0) 0.0]
+   [(< x 0.0) (if (even? n)
+                  (bessj n (- x))
+                  (- (bessj n (- x))))]
+   [(fx< n 0) (if (even? n)
+                  (bessj (- n) x)
+                  (- (bessj (- n) x)))]
+   [else
+    (let* ([ax (magnitude x)]
+           [2/ax (/ 2.0 ax)])
+      (if (> ax n)
+          (let lp ([j 1]
+                   [bjm (bessj₀ ax)]
+                   [bj (bessj₁ ax)])
+            (if (fx= j n)
+                (if (and (< x 0.0) (odd? n)) (- bj) bj)
+                (lp (fx+ j 1)
+                    bj
+                    (- (* j 2/ax bj)
+                       bjm))))
+          (let ([m (round-to-even (+ n (sqrt (* acc n))))]
+                [bj 1.0]
+                [bjp 0.0]
+                [ans 0.0]
+                [sum 0.0])
+            (let lp ([j m])
+              (when (fx= j 0)
+                (let ([ans (/ ans (- (* 2.0 sum) bj))])
+                  (if (and (< x 0.0) (odd? n)) (- ans) ans)
+                  (let ([bjm (- (* j 2/ax bj) bjp)])
+                    (set! bjp bj)
+                    (set! bj bjm)
+                    (when (> (magnitude bj) bigno)
+                      (set! bj  (* bj bigni))
+                      (set! bjp (* bjp bigni))
+                      (set! ans (* ans bigni))
+                      (set! sum (* sum bigni)))
+                    (when (odd? j)
+                      (set! sum (+ sum bj)))
+                    (when (fx= j n)
+                      (set! ans bjp))
+                    (lp (fx- j 1)))))))))]))
+
+(define (fact n)
```

Comment:
Or rename it. `(require (rename-in (only-in math/number-theory factorial) [factorial fact]))`
... that's kinda ugly tho